File: frontend/libs/aas-designer-shared/src/lib/contact/model/request-for-offer.ts
import { Organisation } from '@aas-designer-model';
import { OfferStatus } from './offer-status';

export class RequestForOffer {
  static fromDto(dto: any): any {
    const req = new RequestForOffer();
    Object.assign(req, dto);

    req.price = +dto.price;
    req.firstname = dto.vorname;

    if (dto.organisation != null) req.organisation = Organisation.fromDto(dto.organisation);
    if (dto.anlageDatum != null) req.anlageDatum = new Date(dto.anlageDatum);
    if (dto.aenderungsDatum != null) req.aenderungsDatum = new Date(dto.aenderungsDatum);

    return req;
  }

  id: number | undefined = undefined;

  firstname: string = '';
  name: string = '';
  _email: string = '';
  organisation: Organisation | null = null;
  status: OfferStatus = OfferStatus.New;

  paymentPeriod: 'MONTHLY' | 'YEARLY' = 'MONTHLY';
  numberOfLicences: string = '';
  price: number = 0;
  anlageDatum: Date | undefined;
  aenderungsDatum: Date | undefined;

  set email(value: string | number) {
    this._email = value.toString();
  }

  get email(): string {
    return this._email.toString();
  }

  toDto() {
    return {
      id: this.id,
      name: this.name,
      vorname: this.firstname,
      email: this.email,
      paymentPeriod: this.paymentPeriod,
      numberOfLicences: this.numberOfLicences,
      status: this.status,
      organisation: this.organisation?.toDto(),
    };
  }
}

File: services/aas-designer/AasDesignerModel/Model/RequestForOffer.cs
using System.Text.Json.Serialization;

namespace AasDesignerApi.Model
{
    public class RequestForOffer : IMetadata, IHardDeletable
    {
        public long Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Vorname { get; set; } = string.Empty;
        public string? Email { get; set; } = string.Empty;
        public string? PaymentPeriod { get; set; } = string.Empty;
        public string? NumberOfLicences { get; set; } = string.Empty;
        public string Price { get; set; } = string.Empty;
        public Organisation? Organisation { get; set; }
        public long? OrganisationId { get; set; }

        #region Verwaltungsdaten
        public DateTime AnlageDatum { get; set; } = DateTime.Now;
        public string AnlageBenutzer { get; set; } = string.Empty;
        public DateTime AenderungsDatum { get; set; } = DateTime.Now;
        public string AenderungsBenutzer { get; set; } = string.Empty;
        public bool Geloescht { get; set; } = false;
        public int AenderungsZaehler { get; set; } = 0;

        public long? BesitzerId { get; set; }

        [JsonIgnore]
        public Benutzer? Besitzer { get; set; }
        #endregion

        public OfferStatus Status { get; set; } = OfferStatus.New;
    }

    public enum OfferStatus
    {
        New,
        InProgress,
        SentToCustomer,
        Ordered,
        Rejected,
    }
}
